<a href="https://colab.research.google.com/github/di-yeferson/analitica-empresarial-integrada/blob/main/AEI_LABORATORIO_2_QUISPE__DIEGO.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# LABORATORIO DIRIGIDO N.° 02
## Analítica Empresarial Integrada — TECSUP 2026-II

| | |
|---|---|
| **Docente** | Pilar Rocío Sayán Mejía |
| **Laboratorio** | Perfil analítico comparado de empresas del sector azucarero peruano |
| **N.° de laboratorio** | 2 |
| **Tema** | Estadios de evolución analítica y modelo DELTA de madurez organizacional |
| **Plan curricular** | Big Data y Ciencia de Datos · Código C28 · 2026-II |
| **Duración** | 190 minutos (7:00 p.m. – 10:10 p.m., con receso de 15 minutos) |
| **Fuente de datos** | SMV y BCRP — datos reales y oficiales |
| **Modalidad** |

---

**Equipo N.°:** _____ &nbsp;&nbsp;&nbsp; **Sección:** __2__ &nbsp;&nbsp;&nbsp; **Fecha:** _____

**Apellidos y nombres del estudiante:**  QUISPE HUAMANI, DIEGO YEFERSON
**Integrantes del equipo:**

*   Diego Yeferson Quispe Huamani
*   Renzo Espinoza Tuesta
*   Joel Ronald Nino de Guzman Rodriguez



---


**Elemento de la capacidad terminal:** comprende los estadios de evolución analítica y las dimensiones del modelo DELTA y de su ampliación DELTA Plus; aplica estos marcos a la comparación de indicadores financieros reales y diferencia evidencia, inferencia y opinión.

> **Alcance.** Se analizan estados financieros individuales y anuales del ejercicio 2024. Los importes de la SMV se expresan en miles y pueden estar reportados en soles o dólares según la empresa. Las razones financieras son comparables cuando el numerador y el denominador pertenecen a la misma empresa, moneda y periodo.


## Organización de la sesión — 7:00 p. m. a 10:10 p. m.

**Duración total:** 190 minutos · **Receso:** 15 minutos · **Trabajo efectivo:** 175 minutos.

| Horario | Tiempo | Desarrollo |
|---|---:|---|
| 7:00–7:10 | 10 min | Apertura del caso y conexión con los estadios de evolución analítica. |
| 7:10–7:35 | 25 min | Actividad 1. Revisión de conceptos. Modelos DELTA y DELTA Plus. |
| 7:35–8:00 | 25 min | Actividad 2. Entorno y descarga de información financiera de la SMV. |
| 8:00–8:30 | 30 min | Actividad 2. Normalización, validación e indicadores financieros. Reto 1. |
| 8:30–8:45 | 15 min | **RECESO** |
| 8:45–9:15 | 30 min | Actividad 2. Comparación con Polars y DuckDB. Visualización. Reto 2. |
| 9:15–9:35 | 20 min | Actividad 2. Contexto sectorial del BCRP. Reto 3. |
| 9:35–10:00 | 25 min | **Reto de aplicación y retroalimentación.** Ejercicios 1 a 5. |
| 10:00–10:10 | 10 min | Actividad 3. Informe, conclusiones y preguntas de DELTA Plus. |

> **Regla de trabajo:** no avance de bloque sin registrar la interpretación solicitada. El objetivo no es ejecutar celdas, sino convertir datos en evidencia para una decisión.


## Actividad 1 — Revisión de conceptos (25 minutos)

**Propósito.** Consolidar el vocabulario técnico de la semana antes del desarrollo práctico. La precisión conceptual constituye un requisito del análisis: un indicador correctamente calculado pierde valor si se interpreta con categorías imprecisas.

**Instrucciones.** Complete la tabla con definiciones elaboradas con sus propias palabras. No se admite la reproducción literal de fuentes externas ni de sistemas generativos. La columna de la derecha contiene una pregunta de apoyo: si su definición permite responderla, la definición es suficiente; si no lo permite, corríjala antes de continuar.

**Evidencia esperada.** Tabla completa con las definiciones registradas.

| Concepto | Definición elaborada por el estudiante | Pregunta de apoyo |
|---|---|---|
| Estadios de la evolución analítica | | ¿Qué distingue a una organización del estadio 2 de una del estadio 3? |
| Modelo DELTA | | ¿Por qué el modelo sostiene que las dimensiones deben avanzar de manera conjunta? |
| Modelo DELTA Plus | | ¿Qué limitación del modelo original corrige la ampliación? |
| Dimensión Technology | | ¿Puede una organización con datos y analistas presentar madurez limitada? Justifique. |
| Dimensión Analytical Techniques | | ¿Qué diferencia una técnica descriptiva de una diagnóstica? Proponga un ejemplo. |
| Organización que compite con analítica | | ¿Basta con disponer de tableros de control para competir con analítica? |
| Ventaja competitiva basada en datos | | ¿Por qué razones esta ventaja puede resultar difícil de imitar por un competidor? |
| Impacto de la IA generativa en la empresa | | ¿Qué tarea del análisis financiero no debería delegarse a un sistema generativo? |
| Perfil analítico comparado | | ¿Qué condiciones deben cumplirse para que dos empresas sean comparables? |
| Indicador financiero como evidencia | | ¿En qué caso un indicador correctamente calculado no constituye evidencia válida? |

**Criterio de cierre.** No se avanza al desarrollo práctico mientras existan conceptos sin definición registrada.


## Marco conceptual — Del modelo DELTA al modelo DELTA Plus

El modelo DELTA fue formulado por Davenport, Harris y Morison en *Analytics at Work* (2010) como un instrumento para evaluar la madurez analítica de una organización a partir de cinco dimensiones: **Data**, referida a la disponibilidad, integración y calidad de los datos; **Enterprise**, que examina si la capacidad analítica se encuentra dispersa en áreas aisladas o coordinada a escala institucional; **Leadership**, que valora el compromiso efectivo de la dirección con la decisión basada en evidencia; **Targets**, que analiza si los esfuerzos analíticos se concentran en los objetivos de mayor valor para el negocio; y **Analysts**, que considera la disponibilidad de profesionales con las competencias requeridas. El principio central del modelo establece que estas dimensiones deben progresar de manera conjunta, dado que un avance aislado en una de ellas no eleva la madurez global de la organización.

Con posterioridad, y ante la transformación del entorno tecnológico de la analítica empresarial, el modelo fue ampliado bajo la denominación **DELTA Plus**, que incorpora dos dimensiones adicionales. La primera es **Technology**, que comprende la infraestructura y las plataformas que permiten desarrollar y desplegar capacidades analíticas: soluciones en la nube, *data warehouses*, *data lakes* y arquitecturas *lakehouse*, entornos de procesamiento de grandes volúmenes de información, herramientas de visualización, plataformas analíticas corporativas, componentes de código abierto e infraestructura para el desarrollo de modelos de aprendizaje automático e inteligencia artificial. Su incorporación obedece a una constatación de orden práctico: una organización puede disponer de datos pertinentes y de analistas calificados y presentar, aun así, una madurez analítica limitada si carece de la tecnología necesaria para almacenar, procesar, analizar y poner en uso esa información de manera periódica y reproducible.

La segunda dimensión añadida es **Analytical Techniques**, que evalúa la variedad y la sofisticación de las técnicas empleadas para convertir los datos en información útil para la decisión. Estas técnicas admiten una progresión reconocible: la analítica **descriptiva** establece qué ocurrió; la **diagnóstica** indaga por qué ocurrió; la **predictiva** estima qué podría ocurrir; y la **prescriptiva** determina qué conviene hacer. Sobre esa base se incorporan progresivamente el aprendizaje automático, las técnicas de segmentación, los modelos de clasificación, los pronósticos, los métodos de optimización y las aplicaciones de inteligencia artificial. Dichas técnicas se presentan aquí como ilustración de la evolución de la capacidad analítica de una organización y serán objeto de desarrollo en semanas posteriores del curso; el presente laboratorio se sitúa en los niveles descriptivo y diagnóstico.

### Tabla comparativa

| | Modelo DELTA (2010) | Modelo DELTA Plus |
|---|---|---|
| **Dimensiones** | Cinco | Siete |
| **Componentes** | Data, Enterprise, Leadership, Targets, Analysts | Las cinco anteriores, más Technology y Analytical Techniques |
| **Objeto de evaluación** | Condiciones organizacionales de la analítica | Condiciones organizacionales y capacidad técnica de ejecución |
| **Pregunta que responde** | ¿Dispone la organización de datos, dirección, objetivos y personas? | Además de lo anterior, ¿cuenta con la tecnología y las técnicas para ejecutarlo de manera sostenida? |
| **Limitación que corrige** | — | Una organización con datos y analistas puede presentar madurez limitada por ausencia de tecnología o por el uso de técnicas elementales |

> **Precisión conceptual.** DELTA y DELTA Plus no constituyen modelos alternativos. DELTA Plus es una ampliación del modelo original: conserva íntegramente sus cinco dimensiones y añade dos. Al emitir un diagnóstico debe indicarse explícitamente cuál de los dos marcos se está aplicando.


## Actividad 2 — Desarrollo práctico y ejecución

**Propósito:** descargar información financiera anual y real de empresas del sector azucarero peruano, construir un perfil comparado de rentabilidad, liquidez y endeudamiento, y relacionarlo prudentemente con el contexto sectorial del BCRP mediante pandas, Polars, DuckDB y Plotly.

**Empresas:** Casa Grande S.A.A., Cartavio S.A.A., Agroindustrial Laredo S.A.A. y Empresa Agroindustrial Pomalca S.A.A.


## Paso 1: Preparación del entorno


In [1]:
# Preparación del entorno en Colab. Fijo las versiones a propósito:
# así el notebook corre igual hoy y cuando se vuelva a ejecutar.
%pip install -q polars==1.17.1 duckdb==1.1.3 econdata==1.0.17

import json, html, re, unicodedata
from xml.etree import ElementTree as ET

import requests
import pandas as pd
import numpy as np
import polars as pl
import duckdb
import plotly.graph_objects as go
from IPython.display import display
from econdata import BCRP

# Ensancho la salida de pandas para que las tablas no salgan cortadas.
pd.set_option("display.width", 160)
pd.set_option("display.max_columns", 40)
print("Entorno listo.")


   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 36.4/36.4 MB 36.0 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 20.2/20.2 MB 43.9 MB/s eta 0:00:00
Entorno listo.


## Paso 2: Descarga de información financiera real desde la SMV

Se emplea el servicio web oficial de la SMV para consultar las principales cuentas y el Estado de Situación Financiera. Las operaciones utilizadas son `obtener_InfoFinanciera` y `obtener_BalanceGeneral`. El nombre oficial `obtener_GanciaPerdida` también se conserva para consultas del Estado de Resultados.


In [2]:
SERVICIO_SMV = "https://mvnet.smv.gob.pe/ws_od_eeff/WebServiceInfoFinanciera.asmx"
EJERCICIO = 2024
PERIODO = "A"   # Anual, no trimestral
TIPO = "I"      # Individual: cada empresa por separado, sin consolidar
AZUCARERAS = ["CASA GRANDE", "CARTAVIO", "AGROINDUSTRIAL LAREDO", "POMALCA"]

def etiqueta_local(etiqueta):
    '''Devuelve solo el nombre de la etiqueta, sin el namespace que antepone el XML.'''
    return str(etiqueta).split("}")[-1]

def payload_a_dataframe(payload):
    '''La SMV unas veces responde JSON y otras XML: aquí se cubren ambos casos.'''
    payload = html.unescape(str(payload)).strip()
    try:
        objeto = json.loads(payload)
        if isinstance(objeto, dict):
            objeto = objeto.get("rows", objeto.get("data", objeto))
        if isinstance(objeto, dict):
            objeto = [objeto]
        return pd.DataFrame(objeto)
    except json.JSONDecodeError:
        raiz = ET.fromstring(payload)
        registros = []
        for nodo in raiz.iter():
            hijos = list(nodo)
            if len(hijos) >= 5 and all(not list(hijo) for hijo in hijos):
                registros.append({etiqueta_local(h.tag): h.text for h in hijos})
        return pd.DataFrame(registros)

def descargar_smv(operacion, ejercicio=EJERCICIO, periodo=PERIODO, tipo=TIPO):
    '''Arma el sobre SOAP, llama al servicio de la SMV y verifica que sí traiga registros.'''
    sobre = f'''<?xml version="1.0" encoding="utf-8"?>
    <soap:Envelope xmlns:soap="http://schemas.xmlsoap.org/soap/envelope/">
      <soap:Body><{operacion} xmlns="http://tempuri.org/">
        <Ejercicio>{ejercicio}</Ejercicio><Periodo>{periodo}</Periodo><Tipo>{tipo}</Tipo>
      </{operacion}></soap:Body>
    </soap:Envelope>'''
    respuesta = requests.post(
        SERVICIO_SMV,
        data=sobre.encode("utf-8"),
        timeout=90,
        headers={
            "Content-Type": "text/xml; charset=utf-8",
            "SOAPAction": f'"http://tempuri.org/{operacion}"',
        },
    )
    respuesta.raise_for_status()
    raiz = ET.fromstring(respuesta.content)
    resultado = next(
        (n for n in raiz.iter() if etiqueta_local(n.tag) == f"{operacion}Result"),
        None,
    )
    if resultado is None or not (resultado.text or "").strip():
        raise ValueError(f"La SMV no devolvió datos para {operacion}.")
    datos = payload_a_dataframe(resultado.text)
    if datos.empty:
        raise ValueError(f"La respuesta de {operacion} no contiene registros.")
    return datos

# Dos llamadas: cuentas principales y estado de situación financiera.
principales = descargar_smv("obtener_InfoFinanciera")
balance = descargar_smv("obtener_BalanceGeneral")

print("Principales cuentas:", principales.shape)
print("Estado de situación financiera:", balance.shape)
display(principales.head(3).T)


Principales cuentas: (276, 16)
Estado de situación financiera: (20574, 15)


,0,1,2
RPJ,L00474,I00004,OE7511
TipoEmpresa,EMPRESAS MERCADO ALTERNATIVO DE VALORES,SOCIEDADES ADMINISTRADORAS DE FONDOS DE INVERSION,SOCIEDADES ADMINISTRADORAS DE FONDOS DE INVERSION
TipoSector,DIVERSOS,,
NombreEmpresa,A. JAIME ROJAS REPRESENTACIONES GENERALES S.A.,AC CAPITALES SOCIEDAD ADMINISTRADORA DE FONDOS...,ACRES SOCIEDAD ADMINISTRADORA DE FONDOS DE INV...
RUC,20102032951,20504893295,20601498996
CIIU,5190,6430,6430
Ejercicio,2024,2024,2024
TipoInformacion,Anual Individual,Anual Individual,Anual Individual
Trimestre,Anual,Anual,Anual
Moneda,Soles,Soles,Soles


## Paso 3: Normalización, filtrado y validación del sector

Se normalizan los nombres de las columnas, se filtran las cuatro empresas y se verifica que todas estén representadas. No se utiliza una base simulada ni un respaldo inventado.


In [3]:
# La SMV no siempre escribe igual los nombres de columna, así que primero
# los normalizo (minúsculas, sin tildes) y recién después busco.
def nombre_columna(texto):
    texto = re.sub(r"(?<!^)(?=[A-Z])", "_", str(texto))
    texto = unicodedata.normalize("NFKD", texto).encode("ascii", "ignore").decode()
    return re.sub(r"[^a-z0-9]+", "_", texto.lower()).strip("_")

def preparar(df):
    df = df.copy()
    df.columns = [nombre_columna(c) for c in df.columns]
    return df

def resolver_columna(df, candidatos):
    '''Ubica una columna por su nombre normalizado, aunque cambie el estilo de escritura.'''
    disponibles = {nombre_columna(c): c for c in df.columns}
    for candidato in candidatos:
        clave = nombre_columna(candidato)
        if clave in disponibles:
            return disponibles[clave]
    raise KeyError(f"No se encontró ninguna de estas columnas: {candidatos}. Disponibles: {list(df.columns)}")

def estandarizar_empresa(df):
    df = preparar(df)
    origen = resolver_columna(df, ["nombre_empresa", "razon_social", "empresa"])
    return df.rename(columns={origen: "nombre_empresa"})

def filtrar_azucareras(df):
    patron = "|".join(map(re.escape, AZUCARERAS))
    return df[
        df["nombre_empresa"].astype(str).str.upper().str.contains(patron, regex=True, na=False)
    ].copy()

principales = filtrar_azucareras(estandarizar_empresa(principales))
balance = filtrar_azucareras(estandarizar_empresa(balance))

empresas = sorted(principales["nombre_empresa"].dropna().unique())
print("Empresas encontradas:")
for empresa in empresas:
    print("-", empresa)

# Si alguna de las cuatro no aparece, prefiero que falle acá y no más adelante.
faltantes = [nombre for nombre in AZUCARERAS if not any(nombre in e.upper() for e in empresas)]
if faltantes:
    raise ValueError(f"No se localizaron todas las empresas. Faltantes: {faltantes}")


Empresas encontradas:
- AGROINDUSTRIAL LAREDO S.A.A.
- CARTAVIO SOCIEDAD ANONIMA ABIERTA
- CASA GRANDE SOCIEDAD ANONIMA ABIERTA
- EMPRESA AGROINDUSTRIAL POMALCA S.A.A.


### Paso 3: La cuenta correcta no se busca por su nombre

El Estado de Situación Financiera trae 69 cuentas por empresa, y **más de una contiene el texto «Activos Corrientes»**, pero solo una es el total. Filtrar por la descripción es un error frecuente y silencioso: no lanza ninguna excepción, devuelve un número, y ese número está mal.

Ejecute la celda siguiente y observe el catálogo: la cuenta `1D0122` —«Activos Corrientes Distintos al Efectivo Pignorados como Garantía Colateral»— vale 0 y, de usarse, produce una razón corriente igual a cero.


In [4]:
# Reviso, en una sola empresa, todas las cuentas cuya descripción dice "corriente".
# Es para ver con mis propios ojos por qué filtrar por texto es riesgoso.
muestra = balance[balance["nombre_empresa"] == empresas[0]]
columna_desc = resolver_columna(muestra, ["descripcion_cuenta", "descripcion"])
catalogo = muestra[
    muestra[columna_desc].astype(str).str.contains("orriente", na=False)
][["cuenta", columna_desc, "monto1"]]
print("Empresa de referencia:", empresas[0])
display(catalogo.reset_index(drop=True))

# Por eso declaro los dos totales por su CÓDIGO oficial. El código se mantiene
# entre ejercicios; la descripción cambia, y ahí es donde se cuelan los errores.
TOTAL_ACTIVO_CORRIENTE = "1D01ST"
TOTAL_PASIVO_CORRIENTE = "1D03ST"

print("Códigos declarados:", TOTAL_ACTIVO_CORRIENTE, TOTAL_PASIVO_CORRIENTE)


Empresa de referencia: AGROINDUSTRIAL LAREDO S.A.A.


,cuenta,descripcion_cuenta,monto1
0,1D0122,Activos Corrientes Distintos al Efectivo Pigno...,0.0
1,1D0118,Total Activos Corrientes Distintos de los Acti...,137966.0
2,1D0115,Activos no Corrientes o Grupos de Activos para...,0.0
3,1D01ST,Total Activos Corrientes,137966.0
4,1D0223,"Activos por Impuestos Corrientes, no Corrientes",0.0
5,1D0224,Activos no Corrientes Distintos al Efectivo Pi...,0.0
6,1D02ST,Total Activos No Corrientes,802358.0
7,1D0315,Total de Pasivos Corrientes Distintos de Pasiv...,125303.0
8,1D03ST,Total Pasivos Corrientes,125303.0
9,1D0412,"Pasivos por Impuestos Corrientes, no Corriente",0.0


Códigos declarados: 1D01ST 1D03ST


## Paso 4: Construcción reproducible de los indicadores financieros

| Familia | Indicador | Fórmula |
|---|---|---|
| Rentabilidad | Margen neto | Utilidad neta / Ingresos |
| Rentabilidad | ROA | Utilidad neta / Activo total |
| Rentabilidad | ROE | Utilidad neta / Patrimonio |
| Liquidez | Razón corriente | Activo corriente / Pasivo corriente |
| Endeudamiento | Endeudamiento | Pasivo total / Activo total |
| Endeudamiento | Apalancamiento | Pasivo total / Patrimonio |

La razón corriente se obtiene del Estado de Situación Financiera mediante las descripciones oficiales de activos y pasivos corrientes.


In [5]:
def renombrar_primera(df, destino, candidatos):
    if destino in df.columns:
        return df
    return df.rename(columns={resolver_columna(df, candidatos): destino})

# Para cada cuenta dejo una lista de nombres posibles y me quedo con el primero
# que exista, porque obtener_InfoFinanciera no siempre usa la misma etiqueta.
for destino, candidatos in {
    "ventas": ["total_ingreso", "total_ingresos", "ingresos", "ventas_netas"],
    "utilidad_neta": ["utilidad_neta", "ganancia_perdida_neta", "resultado_neto"],
    "activo_total": ["activo_total", "total_activo", "total_activos"],
    "pasivo_total": ["pasivo_total", "total_pasivo", "total_pasivos"],
    "patrimonio": ["patrimonio_total", "total_patrimonio", "patrimonio"],
}.items():
    principales = renombrar_primera(principales, destino, candidatos)

columnas_base = [
    "nombre_empresa", "ventas", "utilidad_neta", "activo_total",
    "pasivo_total", "patrimonio"
]
perfil = principales[columnas_base].drop_duplicates("nombre_empresa").copy()
perfil = perfil.rename(columns={"nombre_empresa": "empresa"})

def a_numero(valor):
    '''Pasa los importes de la SMV a número: limpia separadores, espacios y paréntesis.'''
    if pd.isna(valor):
        return np.nan
    if isinstance(valor, (int, float, np.number)):
        return float(valor)
    texto = str(valor).strip().replace("\u00a0", "").replace(" ", "")
    negativo = texto.startswith("(") and texto.endswith(")")
    texto = texto.strip("()")
    # La SMV usa punto decimal, así que las comas son solo separador de miles.
    texto = texto.replace(",", "")
    texto = re.sub(r"[^0-9.\-]", "", texto)
    if texto in {"", "-", "."}:
        return np.nan
    numero = float(texto)
    return -numero if negativo else numero

for columna in ["ventas", "utilidad_neta", "activo_total", "pasivo_total", "patrimonio"]:
    perfil[columna] = perfil[columna].map(a_numero)

# Acá se ve la razón del criterio: busco por código y no por descripción.
# La cuenta 1D0122 también dice "Activos Corrientes" pero vale 0, y si la
# tomara por error la razón corriente saldría 0 sin que nada avise del fallo.
def extraer_total_corriente(df, codigo, nombre):
    col_codigo = resolver_columna(df, ["cuenta", "codigo_cuenta", "codigo"])
    monto = resolver_columna(df, ["monto1", "monto", "importe", "valor"])
    datos = df[df[col_codigo].astype(str).str.strip() == codigo][
        ["nombre_empresa", monto]].copy()
    datos[monto] = datos[monto].map(a_numero)
    datos = datos.dropna().drop_duplicates("nombre_empresa")
    if len(datos) < len(AZUCARERAS):
        raise ValueError(
            f"La cuenta {codigo} no está disponible para las cuatro empresas "
            f"en el ejercicio {EJERCICIO}."
        )
    return datos.rename(columns={"nombre_empresa": "empresa", monto: nombre})

perfil = perfil.merge(
    extraer_total_corriente(balance, TOTAL_ACTIVO_CORRIENTE, "activo_corriente"),
    on="empresa", how="inner")
perfil = perfil.merge(
    extraer_total_corriente(balance, TOTAL_PASIVO_CORRIENTE, "pasivo_corriente"),
    on="empresa", how="inner")

if len(perfil) < 4:
    raise ValueError("No fue posible construir el perfil completo de las cuatro empresas.")

columnas_calculo = [
    "ventas", "utilidad_neta", "activo_total", "pasivo_total", "patrimonio",
    "activo_corriente", "pasivo_corriente"
]
faltantes = perfil[columnas_calculo].isna().sum()
if faltantes.any():
    raise ValueError(
        "Existen cuentas sin valor numérico: " +
        ", ".join(f"{c} ({n})" for c, n in faltantes.items() if n)
    )

if (perfil[["ventas", "activo_total", "patrimonio", "pasivo_corriente"]] == 0).any().any():
    raise ValueError("Existe un denominador igual a cero; revise las cuentas antes de calcular.")

# Recién con todo validado calculo los seis indicadores.
perfil["margen_neto"] = perfil["utilidad_neta"] / perfil["ventas"]
perfil["roa"] = perfil["utilidad_neta"] / perfil["activo_total"]
perfil["roe"] = perfil["utilidad_neta"] / perfil["patrimonio"]
perfil["razon_corriente"] = perfil["activo_corriente"] / perfil["pasivo_corriente"]
perfil["endeudamiento"] = perfil["pasivo_total"] / perfil["activo_total"]
perfil["apalancamiento"] = perfil["pasivo_total"] / perfil["patrimonio"]

numeros = perfil.select_dtypes(include="number").to_numpy()
if np.isinf(numeros[~np.isnan(numeros)]).any():
    raise ValueError("Se obtuvo una división entre cero; revise las cuentas de la empresa afectada.")

display(perfil.round(3).T)


,0,1,2,3
empresa,AGROINDUSTRIAL LAREDO S.A.A.,CARTAVIO SOCIEDAD ANONIMA ABIERTA,CASA GRANDE SOCIEDAD ANONIMA ABIERTA,EMPRESA AGROINDUSTRIAL POMALCA S.A.A.
ventas,438934.0,535074.0,637681.0,234045.0
utilidad_neta,55453.0,91846.0,73490.0,9636.0
activo_total,940324.0,926161.0,1923732.0,1352469.0
pasivo_total,327342.0,542687.0,874083.0,679594.0
patrimonio,612982.0,383474.0,1049649.0,672875.0
activo_corriente,137966.0,294933.0,429304.0,168835.0
pasivo_corriente,125303.0,307537.0,362479.0,108618.0
margen_neto,0.126,0.172,0.115,0.041
roa,0.059,0.099,0.038,0.007


**Pregunta 1.** Identifique la empresa con mayor ROE y la empresa con mayor endeudamiento. ¿Coinciden? Interprete la relación observada sin afirmar causalidad.

**Respuesta:**

No coinciden. El ROE más alto lo tiene Cartavio, mientras que Casa Grande es la que sostiene una mayor parte de sus activos con deuda. Lo que se ve es que cada una llega a su resultado por un camino distinto: una se apoya más en financiamiento externo y la otra consigue un mejor retorno sobre el patrimonio de sus accionistas. Es una relación descriptiva, no causal: con estos datos no se puede afirmar que un indicador explique al otro.

---


### Reto 1 — Predicción antes del ranking (5 minutos)

**Sin ordenar todavía la tabla**, responda en equipo:

1. ¿Qué empresa creen que presenta el mejor perfil financiero?
    - Cartavio S.A.A.
2. Sustenten la predicción con **dos indicadores** como mínimo.
    - Es la que tiene el margen neto más alto (0.172) y también el ROE más alto (0.24).
3. ¿Qué indicador podría hacer cambiar su conclusión?
    - La razón corriente. Si Cartavio tuviera una liquidez baja frente a las demás, se le complicaría cubrir sus obligaciones de corto plazo y ese "mejor perfil" ya no sería tan claro.

**Predicción del equipo:**

**Justificación:**

Después de ejecutar el ranking del Paso 5, vuelva a esta respuesta y escriba: **¿se confirmó o no la predicción y por qué?**

- Sí se confirmó: al ordenar la tabla por ROE, Cartavio queda en el primer lugar.


## Paso 5: Comparación sectorial con Polars y DuckDB


In [6]:
# El mismo perfil, ahora en Polars, ordenado de mayor a menor ROE.
perfil_pl = pl.from_pandas(perfil)

ranking_roe = (
    perfil_pl
    .select([
        "empresa", "margen_neto", "roa", "roe", "razon_corriente",
        "endeudamiento", "apalancamiento"
    ])
    .sort("roe", descending=True)
)
ranking_roe


empresa,margen_neto,roa,roe,razon_corriente,endeudamiento,apalancamiento
str,f64,f64,f64,f64,f64,f64
"""CARTAVIO SOCIEDAD ANONIMA ABIE…",0.171651,0.099169,0.23951,0.959016,0.585953,1.415186
"""AGROINDUSTRIAL LAREDO S.A.A.""",0.126336,0.058972,0.090464,1.101059,0.348116,0.534016
"""CASA GRANDE SOCIEDAD ANONIMA A…",0.115246,0.038202,0.070014,1.184356,0.454368,0.832738
"""EMPRESA AGROINDUSTRIAL POMALCA…",0.041172,0.007125,0.014321,1.554392,0.502484,1.009986


In [7]:
# La misma pregunta del ranking anterior, pero escrita en SQL sobre DuckDB.
# Si dos motores distintos devuelven el mismo orden, el resultado es
# reproducible y no una casualidad del método usado.
CONSULTA = """
SELECT empresa,
       ROUND(margen_neto * 100, 1)    AS margen_neto_pct,
       ROUND(roa * 100, 1)            AS roa_pct,
       ROUND(roe * 100, 1)            AS roe_pct,
       ROUND(razon_corriente, 2)      AS razon_corriente,
       ROUND(endeudamiento * 100, 1)  AS endeudamiento_pct,
       ROUND(apalancamiento, 2)       AS apalancamiento
FROM perfil
ORDER BY roe_pct DESC
"""

consulta = duckdb.sql(CONSULTA).df()
display(consulta)

# Lo verifico con código en vez de compararlo a ojo.
if list(consulta["empresa"]) == list(ranking_roe["empresa"]):
    print("Correcto: DuckDB y Polars devuelven el mismo ordenamiento.")
else:
    print("Los ordenamientos no coinciden. Revise las columnas y el ORDER BY.")


,empresa,margen_neto_pct,roa_pct,roe_pct,razon_corriente,endeudamiento_pct,apalancamiento
0,CARTAVIO SOCIEDAD ANONIMA ABIERTA,17.2,9.9,24.0,0.96,58.6,1.42
1,AGROINDUSTRIAL LAREDO S.A.A.,12.6,5.9,9.0,1.10,34.8,0.53
2,CASA GRANDE SOCIEDAD ANONIMA ABIERTA,11.5,3.8,7.0,1.18,45.4,0.83
3,EMPRESA AGROINDUSTRIAL POMALCA S.A.A.,4.1,0.7,1.4,1.55,50.2,1.01


Correcto: DuckDB y Polars devuelven el mismo ordenamiento.


In [ ]:
# Tres rankings a la vez para ver cuánto se mueve cada empresa
# según el criterio con el que se la mire.
ranking_comparado = pd.DataFrame({
    "empresa": perfil["empresa"],
    "rank_roe": perfil["roe"].rank(ascending=False, method="min").astype(int),
    "rank_liquidez": perfil["razon_corriente"].rank(ascending=False, method="min").astype(int),
    # En endeudamiento, menos deuda es mejor posición; es solo un criterio descriptivo de exposición.
    "rank_endeudamiento": perfil["endeudamiento"].rank(ascending=True, method="min").astype(int),
})
ranking_comparado["cambio_max_posicion"] = (
    ranking_comparado[["rank_roe", "rank_liquidez", "rank_endeudamiento"]].max(axis=1)
    - ranking_comparado[["rank_roe", "rank_liquidez", "rank_endeudamiento"]].min(axis=1)
)
display(ranking_comparado.sort_values(["rank_roe", "empresa"]))


,empresa,rank_roe,rank_liquidez,rank_endeudamiento,cambio_max_posicion
1,CARTAVIO SOCIEDAD ANONIMA ABIERTA,1,4,4,3
0,AGROINDUSTRIAL LAREDO S.A.A.,2,3,1,2
2,CASA GRANDE SOCIEDAD ANONIMA ABIERTA,3,2,2,1
3,EMPRESA AGROINDUSTRIAL POMALCA S.A.A.,4,1,3,3


### Reto 2 — Evidencia, inferencia y conclusión (5 minutos)

Clasifique cada afirmación como **Sustentada**, **No necesariamente** o **No puede concluirse** y justifique con cifras:

| Afirmación | Clasificación | Justificación con evidencia |
|---|---|---|
| La empresa con mayor ROE es necesariamente la más competitiva. | No necesariamente |  |
| Una empresa con mayor endeudamiento posee necesariamente menor madurez analítica. | No puede concluirse |  |
| Obtener un ordenamiento consistente con Polars y DuckDB fortalece la reproducibilidad del análisis. | Sustentada |  |

**Importante:** reproducibilidad no significa causalidad ni convierte un indicador financiero en evidencia directa de madurez organizacional.


**Pregunta 2.** Compare los rankings por rentabilidad, liquidez y endeudamiento. Explique qué revela el cambio de posición sobre el perfil competitivo de cada empresa.

**Respuesta:**

Cartavio encabeza la rentabilidad, pero en liquidez las que salen mejor paradas son Casa Grande y Pomalca, y en endeudamiento Casa Grande aparece como la más expuesta. Que las posiciones se muevan tanto según el criterio muestra que ninguna domina en todo: cada empresa tiene una fortaleza distinta y el perfil competitivo recién se entiende leyendo los tres rankings juntos, no uno por separado.

---



## Paso 6: Visualización del perfil comparado


In [8]:
# Primer gráfico: los cuatro indicadores que se expresan en porcentaje.
indicadores_pct = ["margen_neto", "roa", "roe", "endeudamiento"]
nombres_indicadores = {
    "margen_neto": "Margen neto",
    "roa": "ROA",
    "roe": "ROE",
    "endeudamiento": "Endeudamiento",
}
perfil_grafico = perfil.copy()
perfil_grafico["empresa_corta"] = (
    perfil_grafico["empresa"].str.upper()
    .map(lambda x: "Laredo" if "LAREDO" in x else
                   "Cartavio" if "CARTAVIO" in x else
                   "Casa Grande" if "CASA GRANDE" in x else
                   "Pomalca" if "POMALCA" in x else x.title())
)
fig = go.Figure()
for indicador in indicadores_pct:
    fig.add_trace(go.Bar(
        name=nombres_indicadores[indicador],
        x=perfil_grafico["empresa_corta"],
        y=perfil_grafico[indicador] * 100,
        text=(perfil_grafico[indicador] * 100).round(1),
        texttemplate="%{text:.1f}%",
        textposition="outside",
        customdata=perfil_grafico[["empresa"]],
        hovertemplate="%{customdata[0]}<br>%{fullData.name}: %{y:.2f}%<extra></extra>",
    ))
fig.update_layout(
    title=f"Perfil financiero comparado — sector azucarero ({EJERCICIO})",
    barmode="group",
    yaxis_title="Porcentaje (%)",
    height=540,
    width=1050,
    template="plotly_white",
    margin=dict(t=110, b=90, l=80, r=40),
    legend=dict(orientation="h", yanchor="bottom", y=1.02, xanchor="left", x=0),
)
fig.show()

# La razón corriente va en un gráfico aparte: no es porcentaje, se mide en veces.
fig_liquidez = go.Figure(go.Bar(
    x=perfil_grafico["empresa_corta"],
    y=perfil_grafico["razon_corriente"],
    marker_color="#00A9CE",
    text=perfil_grafico["razon_corriente"].round(2),
    texttemplate="%{text:.2f}",
    textposition="outside",
    customdata=perfil_grafico[["empresa"]],
    hovertemplate="%{customdata[0]}<br>Razón corriente: %{y:.2f}<extra></extra>",
))
fig_liquidez.update_layout(
    title="Razón corriente por empresa",
    yaxis_title="Veces",
    height=460,
    width=900,
    template="plotly_white",
    margin=dict(t=80, b=80, l=80, r=40),
)
fig_liquidez.update_yaxes(rangemode="tozero")
fig_liquidez.show()


## Paso 7: Contexto sectorial con datos específicos del BCRP

Se emplean series oficiales sobre tipo de cambio, producción de caña y exportaciones de azúcar. Estas series describen el entorno; por sí solas no demuestran la causa de las diferencias financieras entre empresas.


In [9]:
# Series oficiales del BCRP que describen el entorno del sector, no a las empresas.
SERIES = {
    "PN01209PM": "tipo_cambio",
    "PN01790AM": "produccion_cana_azucar_miles_t",
    "PN38774BM": "export_azucar_fob_millones_usd",
    "PN38775BM": "volumen_exportado_azucar",
}

contexto = BCRP.get_data(SERIES, "2020-01", "2024-12")
print("Datos del BCRP:", contexto.shape)
display(contexto.tail(12))


Datos del BCRP: (60, 4)


,tipo_cambio,produccion_cana_azucar_miles_t,export_azucar_fob_millones_usd,volumen_exportado_azucar
2024-01-01,3.743667,755.602441,7.547009,10.321329
2024-02-01,3.830905,814.525062,4.398561,5.993531
2024-03-01,3.712947,563.626363,1.911468,2.355260
2024-04-01,3.718409,634.675782,3.839971,4.951844
2024-05-01,3.735273,812.740966,3.235732,3.843619
2024-06-01,3.789526,830.637431,0.481814,0.618096
2024-07-01,3.769500,840.996455,0.919740,1.246593
2024-08-01,3.745550,1017.287110,1.010892,1.358606
2024-09-01,3.772143,930.472111,3.097093,4.088378
2024-10-01,3.757500,1053.902968,1.906427,2.543543


In [10]:
# Las series vienen en unidades distintas (soles, toneladas, millones de dólares).
# Las llevo a base 100 solo para poder comparar tendencias; los datos no se modifican.
contexto_grafico = contexto.copy()
columnas_numericas = contexto_grafico.select_dtypes(include="number").columns
base = contexto_grafico[columnas_numericas].apply(lambda s: s.dropna().iloc[0])
indice_100 = contexto_grafico[columnas_numericas].divide(base).multiply(100)

fig_contexto = go.Figure()
for columna in indice_100.columns:
    fig_contexto.add_trace(go.Scatter(
        x=indice_100.index,
        y=indice_100[columna],
        mode="lines",
        name=columna,
    ))
fig_contexto.update_layout(
    title="Contexto del sector azucarero — índice base 100",
    yaxis_title="Índice",
    xaxis_title="Periodo",
    height=480,
)
fig_contexto.show()


**Pregunta 3.** Describa la evolución del tipo de cambio, la producción de caña y las exportaciones de azúcar. Relacione ese contexto con el perfil financiero y diferencie expresamente:

- **Evidencia:** resultado directamente observado.
- **Inferencia:** interpretación razonada basada en la evidencia.
- **Opinión:** valoración que debe justificarse.

**Respuesta:**

- **Evidencia:** durante 2024 el tipo de cambio se mantuvo estable, moviéndose en un rango estrecho de S/ 3.71 a S/ 3.83 por dólar y sin marcar una tendencia clara. La producción de caña sí creció, aunque con altibajos: partió de 564 mil toneladas en marzo, superó las 1,000 mil toneladas en agosto y octubre, y cerró el año en 940 mil. Las exportaciones de azúcar (FOB) hicieron el camino contrario: cayeron desde US$ 7.5 millones en enero hasta mínimos de US$ 0.5–1 millón entre junio y agosto, con una recuperación parcial de US$ 2.1–3.5 millones en noviembre y diciembre. El volumen exportado siguió el mismo comportamiento.

- **Inferencia:** que la producción suba en la segunda mitad del año y las exportaciones no acompañen en la misma proporción sugiere que buena parte de esa caña se habría quedado en el mercado interno o en otros usos, en lugar de salir al exterior. Y como el tipo de cambio casi no se movió, es poco probable que las diferencias de resultados entre las cuatro empresas en 2024 se expliquen por el factor cambiario.

- **Opinión:** es razonable pensar que la caída de exportaciones del primer semestre haya presionado los ingresos de las empresas más orientadas al mercado externo, pero eso habría que contrastarlo con el detalle de ventas internas y externas de cada una, información que los estados financieros analizados no incluyen. Por eso cualquier vínculo directo entre este contexto y el perfil de Casa Grande, Cartavio, Laredo o Pomalca se queda en hipótesis y no en conclusión.
---



### Reto 3 — Diagnóstico provisional de madurez y DELTA (8 minutos)

Seleccione **una** de las cuatro empresas. No se busca “adivinar” su madurez: se busca reconocer qué puede y qué no puede sostenerse con la evidencia disponible.

| Aspecto | Respuesta del equipo |
|---|---|
| Empresa seleccionada | Cartavio Sociedad Anónima Abierta |
| Evidencia financiera observada | Es la más rentable del grupo (ROE 24.0% y margen neto 17.2%), pero con una razón corriente de 0.96 —por debajo de 1— y el endeudamiento más alto de las cuatro (58.6%). |
| Estadio de evolución analítica compatible **de manera provisional** | Estadio inicial-intermedio: lo más probable es que el uso analítico esté concentrado en áreas puntuales y no extendido a toda la organización. Es una clasificación tentativa, sostenida únicamente en resultados financieros. |
| Dimensión DELTA evaluable parcialmente | Targets. Que la rentabilidad sea tan buena podría indicar que sus esfuerzos están puestos en objetivos de alto valor, como margen y eficiencia, pero eso sigue siendo una inferencia y no una comprobación. |
| Dimensiones DELTA que no pueden verificarse | Enterprise, Leadership, Analysts y Technology. Ninguna se puede confirmar leyendo estados financieros, porque exigen mirar procesos internos, equipos e infraestructura de la empresa. |
| Evidencia organizacional adicional requerida | Entrevistas o documentación sobre qué herramientas analíticas usan, cómo está formado su equipo de datos y si hay decisiones estratégicas tomadas a partir de análisis y no solo de resultados contables. |

**Cierre del reto:** redacte una frase que distinga claramente **evidencia**, **inferencia** y **opinión**.



### Pausa metodológica — ¿qué NO podemos afirmar?

Los estados financieros permiten comparar resultados y estructura financiera, pero **no prueban por sí solos** liderazgo analítico, cultura de datos, disponibilidad de analistas, integración empresarial de la analítica ni uso de IA generativa. Por ello, cualquier estadio o perfil DELTA derivado de este laboratorio debe presentarse como **hipótesis provisional**, no como diagnóstico organizacional definitivo.


## Actividad 3 — Informe de perfil analítico comparado

Responda en términos de negocio y sustente cada afirmación con las cifras obtenidas.

**A.** Interprete tres hallazgos del perfil comparado y explique, para cada uno, su implicancia competitiva en el sector.

**Respuesta:**

- Cartavio junta el mejor margen neto (17.2%) con la mejor rotación de activos (0.58), y DuPont lo confirma: de ahí sale el ROE más alto del sector (24.0%). Implicancia: su ventaja hoy pasa por eficiencia operativa y control de costos, no por tamaño.

- Laredo tiene el endeudamiento más bajo (34.8%) y aun así un ROE sólido (9.0%). Implicancia: le queda margen financiero para invertir o aguantar un golpe externo sin comprometer su solvencia, una posición cómoda frente a la volatilidad que mostró el sector exportador en 2024.

- Pomalca combina el ROE más bajo (1.4%) con un endeudamiento alto para su nivel de rendimiento (50.2%) y la mayor proporción de activos no corrientes (87.5%). Implicancia: tiene el capital muy amarrado al largo plazo y con poco retorno, lo que apunta a poca generación de caja en el corto plazo y a una posición competitiva frágil.

---

**B.** Determine si los estados financieros permiten ubicar de manera concluyente a cada empresa en uno de los cinco estadios de evolución analítica. Formule, cuando corresponda, una **clasificación provisional** y precise qué evidencia organizacional adicional se necesita para confirmarla.

**Respuesta:** No lo permiten. Los estados financieros miden resultados económicos, no capacidades organizacionales como liderazgo, cultura de datos, tecnología o talento analítico. Lo máximo que se puede sostener como clasificación provisional es un estadio inicial-intermedio para las cuatro, y por inferencia indirecta desde sus resultados. Para confirmarlo haría falta evidencia organizacional directa: entrevistas a la gerencia, un inventario de las herramientas tecnológicas que usan y casos concretos de decisiones estratégicas tomadas a partir de datos.

---

**C.** A partir del modelo DELTA, ¿qué dimensión resulta más difícil de evaluar únicamente con estados financieros? Explique por qué y qué información adicional se requeriría.

**Respuesta:** La más difícil es Leadership, y muy de cerca Analysts. El compromiso real de la dirección con decidir en base a evidencia no queda registrado en un balance ni en un estado de resultados. Para evaluarla se necesitaría información cualitativa: actas de directorio, entrevistas a la gerencia o señales de inversión sostenida en capacidades analíticas a lo largo del tiempo.

---

**D.** Formule dos recomendaciones sustentadas exclusivamente en los datos observados, dirigidas a la empresa con el perfil competitivo más débil.

**Respuesta:**

- Revisar la composición de sus activos no corrientes, que representan el 87.5% del activo total, para identificar si hay activos subutilizados o de baja rotación que se puedan optimizar o desinvertir y así liberar capital.

- Atacar primero el margen neto, que con 4.1% es el más bajo del grupo, antes que buscar crecer en escala: su rotación de activos (0.17) también es la menor del sector, así que el problema no es solo cuánto vende sino cuánto le queda por cada venta.

---

**E.** Explique cómo la inteligencia artificial generativa podría modificar el análisis de competitividad. Distinga entre usos plausibles y afirmaciones no verificables.

**Respuesta:** Como uso plausible, la IA generativa puede acelerar tareas concretas: resumir estados financieros, escribir el código que calcula indicadores de forma repetible o ayudar a detectar anomalías en catálogos de cuentas muy grandes. Lo que no puede hacer es verificar por sí sola que los datos de origen sean correctos, ni sostener juicios de causalidad financiera o de madurez organizacional sin evidencia adicional. Usarla para "concluir" quién es más competitivo sin números que lo respalden sería una afirmación no verificable.



### Matriz de síntesis para la decisión

Complete esta matriz antes de redactar las conclusiones finales:

| Empresa | Fortaleza principal | Riesgo/debilidad principal | Evidencia numérica | Implicancia competitiva |
|---|---|---|---|---|
| Casa Grande | Liquidez razonable | Rentabilidad baja pese a ser la más grande en activos | ROE 7.0%, ROA 3.8%, activo total 1,923,732 (el mayor del sector) | Su tamaño no se traduce en eficiencia; hay espacio de mejora en la gestión de sus activos |
| Cartavio | La más rentable del sector | Más deuda y menos liquidez que el resto | ROE 24.0%, margen neto 17.2%, endeudamiento 58.6%, razón corriente 0.96 | Lidera en rentabilidad, pero cargando un riesgo financiero mayor |
| Agroindustrial Laredo | Poca deuda con rentabilidad sólida | Rotación intermedia, sin ser la más eficiente |Endeudamiento 34.8%, ROE 9.0%, rotación 0.47 | El equilibrio riesgo-rentabilidad más sano del grupo |
| Pomalca | La mayor liquidez del grupo | Rentabilidad casi nula y capital muy concentrado en el largo plazo | ROE 1.4%, margen neto 4.1%, 87.5% de activos no corrientes, razón corriente 1.55 | La posición competitiva más débil: capital inmovilizado que rinde muy poco |

Luego responda: **¿qué empresa requiere mayor atención gerencial según los indicadores observados y qué dos acciones recomendaría sin exceder la evidencia disponible?**

**Respuesta:**

Pomalca es la que necesita atención con más urgencia: tiene el ROE más bajo del sector (1.4%), el margen neto más bajo (4.1%) y la mayor parte de su capital inmovilizado en el largo plazo (87.5%). Esa combinación describe activos que producen muy poco.

Dos acciones que se sostienen solo con lo observado:

- Evaluar qué tan eficientes son sus activos no corrientes, porque un margen y una rotación tan bajos (0.17) apuntan a capacidad instalada que no está siendo aprovechada frente a las ventas que genera.

- Revisar su estructura de costos operativos: su margen neto (4.1%) queda muy por debajo del resto aun teniendo la mejor razón corriente del grupo (1.55), lo que descarta que el problema principal sea de liquidez inmediata.


---

## Reto de aplicación y retroalimentación

En esta sección se aplicarán los procedimientos desarrollados durante la sesión a nuevas situaciones de análisis. Cada ejercicio requiere modificar, completar o construir código a partir de los datos financieros previamente procesados. Posteriormente, los resultados obtenidos deberán interpretarse brevemente desde una perspectiva empresarial. El propósito es comprobar la comprensión de las técnicas utilizadas y fortalecer la capacidad de adaptar el análisis ante nuevas preguntas de negocio.

**Indicaciones generales.** Los cinco ejercicios operan sobre `perfil_reto`, copia de trabajo del perfil financiero construido en la Actividad 2. Se emplea una copia para que ninguna modificación introducida en esta sección altere los resultados obtenidos con anterioridad. Los datos proceden íntegramente de la SMV; no corresponde generar ni sustituir valores en ningún caso. Cada respuesta escrita no debe exceder cuatro líneas.

**Duración en sesión:** 25 minutos. Los ejercicios que no concluyan durante la sesión se completan como avance del entregable y se depositan en el repositorio del equipo junto con el notebook ejecutado.


In [11]:
# Trabajo sobre una copia para no alterar el perfil de la Actividad 2.
perfil_reto = perfil.copy()
print("Copia de trabajo creada:", perfil_reto.shape)
display(perfil_reto[["empresa", "ventas", "activo_total", "margen_neto", "roa", "roe"]].round(3))


Copia de trabajo creada: (4, 14)


,empresa,ventas,activo_total,margen_neto,roa,roe
0,AGROINDUSTRIAL LAREDO S.A.A.,438934.0,940324.0,0.126,0.059,0.090
1,CARTAVIO SOCIEDAD ANONIMA ABIERTA,535074.0,926161.0,0.172,0.099,0.240
2,CASA GRANDE SOCIEDAD ANONIMA ABIERTA,637681.0,1923732.0,0.115,0.038,0.070
3,EMPRESA AGROINDUSTRIAL POMALCA S.A.A.,234045.0,1352469.0,0.041,0.007,0.014


### Ejercicio 1 — Recuperación de una nueva cuenta financiera del catálogo oficial

Durante la Actividad 2 se identificaron dos cuentas del Estado de Situación Financiera a partir de su **código oficial** y no de su descripción textual. Ese procedimiento constituye el criterio de trazabilidad del laboratorio, dado que el código permanece estable entre ejercicios mientras que la descripción admite variantes que conducen a errores silenciosos.

En este ejercicio deberá replicar el mismo procedimiento sobre una cuenta que aún no ha sido utilizada: el **Total de Activos No Corrientes**. Esta cuenta reúne los recursos de largo plazo de la empresa —terrenos, plantaciones, maquinaria e instalaciones industriales— y su participación sobre el activo total describe la estructura de inversión de cada empresa azucarera.

**Se solicita:**

1. Localizar en el catálogo de la SMV la cuenta correspondiente al Total de Activos No Corrientes e identificar su código oficial.
2. Recuperar el valor de esa cuenta para las cuatro empresas analizadas.
3. Incorporar la variable a `perfil_reto`.
4. Calcular el porcentaje que dicha cuenta representa respecto del activo total.

**Tiempo estimado:** 10 a 12 minutos.


In [12]:
# Paso A: reviso el catálogo para ubicar cuál es el código del total no corriente.
muestra_e1 = balance[balance["nombre_empresa"] == empresas[0]]
columna_desc = resolver_columna(muestra_e1, ["descripcion_cuenta", "descripcion"])
catalogo_no_corriente = muestra_e1[
    muestra_e1[columna_desc].astype(str).str.contains("o Corrientes", case=False, na=False)
][["cuenta", columna_desc, "monto1"]]
display(catalogo_no_corriente.reset_index(drop=True))

,cuenta,descripcion_cuenta,monto1
0,1D0118,Total Activos Corrientes Distintos de los Acti...,137966.0
1,1D0115,Activos no Corrientes o Grupos de Activos para...,0.0
2,1D0223,"Activos por Impuestos Corrientes, no Corrientes",0.0
3,1D0224,Activos no Corrientes Distintos al Efectivo Pi...,0.0
4,1D02ST,Total Activos No Corrientes,802358.0
5,1D04ST,Total Pasivos No Corrientes,202039.0


In [13]:
# Paso B: ya ubicado arriba, lo declaro y con él traigo la cuenta de las cuatro empresas.
TOTAL_ACTIVO_NO_CORRIENTE = "1D02ST"

def obtener_cuenta(df, codigo, columna_valor="monto1"):
    col_cuenta = resolver_columna(df, ["cuenta"])
    resultado = df[df[col_cuenta] == codigo][["nombre_empresa", columna_valor]].copy()
    resultado = resultado.rename(columns={"nombre_empresa": "empresa", columna_valor: "activo_no_corriente"})
    resultado["activo_no_corriente"] = resultado["activo_no_corriente"].map(a_numero)
    return resultado

activo_no_corriente = obtener_cuenta(balance, TOTAL_ACTIVO_NO_CORRIENTE)
perfil_reto = perfil_reto.merge(activo_no_corriente, on="empresa", how="left")
perfil_reto["pct_activo_no_corriente"] = perfil_reto["activo_no_corriente"] / perfil_reto["activo_total"]

display(perfil_reto[["empresa", "activo_no_corriente", "activo_total", "pct_activo_no_corriente"]].round(3))

,empresa,activo_no_corriente,activo_total,pct_activo_no_corriente
0,AGROINDUSTRIAL LAREDO S.A.A.,802358.0,940324.0,0.853
1,CARTAVIO SOCIEDAD ANONIMA ABIERTA,631228.0,926161.0,0.682
2,CASA GRANDE SOCIEDAD ANONIMA ABIERTA,1494428.0,1923732.0,0.777
3,EMPRESA AGROINDUSTRIAL POMALCA S.A.A.,1183634.0,1352469.0,0.875


**Pregunta 4.** ¿Qué empresa presenta la mayor proporción de activos no corrientes y qué podría indicar esto sobre su estructura de inversión?

**Respuesta (máximo cuatro líneas):** La mayor proporción la tiene Empresa Agroindustrial Pomalca S.A.A. con 87.5% del activo total, y muy cerca Agroindustrial Laredo con 85.3%. Eso describe una inversión muy concentrada en el largo plazo —tierras, plantaciones, maquinaria— y por lo tanto menos recursos disponibles en el corto plazo. Cartavio es la más balanceada del grupo (68.2%), lo que podría darle más flexibilidad operativa frente a las demás.

---



### Ejercicio 2 — Construcción de un indicador de eficiencia en el uso de los activos

Los indicadores calculados en la Actividad 2 describen rentabilidad, liquidez y endeudamiento, pero ninguno mide la eficiencia con que la empresa emplea su base de activos para generar ventas. La rotación de activos cubre esa dimensión: expresa cuántas unidades monetarias de venta produce cada unidad monetaria invertida en activos, y se obtiene dividiendo las ventas entre el activo total.

En el sector azucarero este indicador resulta particularmente informativo, por tratarse de empresas intensivas en capital, con extensas superficies de cultivo e instalaciones industriales. Dos empresas pueden alcanzar rentabilidades semejantes mediante estrategias opuestas: una sostenida en márgenes elevados y otra en un mayor volumen de operación sobre la misma base de activos.

**Se solicita:**

1. Calcular la rotación de activos e incorporarla a `perfil_reto` en la columna `rotacion_activos`.
2. Generar un ranking de las empresas ordenado de mayor a menor según dicho indicador.
3. Comparar ese ordenamiento con el obtenido a partir del ROA calculado en la Actividad 2.

**Tiempo estimado:** 10 a 12 minutos.


In [14]:
# Rotación: cuántos soles de venta genera cada sol invertido en activos.
perfil_reto["rotacion_activos"] = perfil_reto["ventas"] / perfil_reto["activo_total"]

ranking_rotacion = perfil_reto[["empresa", "rotacion_activos"]].sort_values(
    "rotacion_activos", ascending=False
)
print("Ranking por rotación de activos:")
display(ranking_rotacion)

ranking_roa_e2 = perfil_reto[["empresa", "roa"]].sort_values("roa", ascending=False)
print("Ranking por ROA (referencia de la Actividad 2):")
display(ranking_roa_e2)

Ranking por rotación de activos:


,empresa,rotacion_activos
1,CARTAVIO SOCIEDAD ANONIMA ABIERTA,0.577733
0,AGROINDUSTRIAL LAREDO S.A.A.,0.466790
2,CASA GRANDE SOCIEDAD ANONIMA ABIERTA,0.331481
3,EMPRESA AGROINDUSTRIAL POMALCA S.A.A.,0.173050


Ranking por ROA (referencia de la Actividad 2):


,empresa,roa
1,CARTAVIO SOCIEDAD ANONIMA ABIERTA,0.099169
0,AGROINDUSTRIAL LAREDO S.A.A.,0.058972
2,CASA GRANDE SOCIEDAD ANONIMA ABIERTA,0.038202
3,EMPRESA AGROINDUSTRIAL POMALCA S.A.A.,0.007125


**Pregunta 5.** ¿La empresa con mayor rotación de activos presenta también el mayor ROA? Explique brevemente a partir de los resultados obtenidos.

**Respuesta (máximo cuatro líneas):** Sí. Cartavio encabeza los dos indicadores: 0.58 veces de rotación y 9.9% de ROA. De hecho el orden de las cuatro empresas es idéntico en ambos rankings (Cartavio > Laredo > Casa Grande > Pomalca), lo que apunta a que aquí el ROA se explica sobre todo por qué tan bien se usan los activos para generar ventas, más que por diferencias grandes de margen entre ellas.

---



### Ejercicio 3 — Verificación numérica de la descomposición DuPont del ROA

La descomposición DuPont establece que la rentabilidad sobre activos puede expresarse como el producto de dos componentes: el margen neto, que mide la rentabilidad obtenida por cada unidad vendida, y la rotación de activos, que mide la intensidad con que se emplea la base de activos.

El valor analítico de esta descomposición no reside en la identidad algebraica, sino en la explicación que aporta: permite establecer si una rentabilidad elevada proviene del precio y del control de costos, del volumen de operación, o de la concurrencia de ambos factores. Corresponde comprobar numéricamente la identidad sobre los datos reales de la SMV, dado que toda relación teórica empleada como evidencia debe verificarse antes de sustentar una conclusión.

**Se solicita:**

1. Construir la columna `roa_dupont` como el producto del margen neto por la rotación de activos.
2. Construir la columna `diferencia_roa` como la diferencia entre el ROA obtenido en la Actividad 2 y `roa_dupont`.
3. Verificar numéricamente la magnitud de esa diferencia y determinar si resulta admisible, considerando el redondeo de los importes reportados por la SMV.

**Tiempo estimado:** 10 a 12 minutos.


In [15]:
# DuPont dice que ROA = margen × rotación. Lo compruebo con los datos reales
# en lugar de darlo por cierto solo porque es una fórmula conocida.
perfil_reto["roa_dupont"] = perfil_reto["margen_neto"] * perfil_reto["rotacion_activos"]
perfil_reto["diferencia_roa"] = perfil_reto["roa"] - perfil_reto["roa_dupont"]

display(perfil_reto[["empresa", "roa", "roa_dupont", "diferencia_roa"]].round(6))
print("Diferencia máxima observada:", perfil_reto["diferencia_roa"].abs().max())

,empresa,roa,roa_dupont,diferencia_roa
0,AGROINDUSTRIAL LAREDO S.A.A.,0.058972,0.058972,-0.0
1,CARTAVIO SOCIEDAD ANONIMA ABIERTA,0.099169,0.099169,0.0
2,CASA GRANDE SOCIEDAD ANONIMA ABIERTA,0.038202,0.038202,0.0
3,EMPRESA AGROINDUSTRIAL POMALCA S.A.A.,0.007125,0.007125,0.0


Diferencia máxima observada: 6.938893903907228e-18


**Pregunta 6.** ¿Qué componente parece explicar mejor el ROA de la empresa líder: el margen neto, la rotación de activos o ambos?

**Respuesta (máximo cuatro líneas):** Para responderlo con precisión calculé el margen implícito de cada empresa (margen = ROA / rotación):

*   Cartavio: 0.0992 / 0.5777 ≈ 17.2%
*   Laredo: 0.0590 / 0.4668 ≈ 12.6%
*   Casa Grande: 0.0382 / 0.3315 ≈ 11.5%
*   Pomalca: 0.0071 / 0.1731 ≈ 4.1%

Los dos componentes juegan a favor de Cartavio, pero el que más la separa es el margen: 17.2% frente al 12.6% de Laredo, su competidor más cercano, que en rotación está casi a la par (0.47 contra 0.58). Es decir, su liderazgo en ROA viene principalmente de un mejor control de costos y precio, y solo en segundo lugar de la eficiencia en el uso de activos.


---



### Ejercicio 4 — Consulta empresarial con condiciones múltiples en DuckDB

En la Actividad 2 se empleó DuckDB para reproducir un ordenamiento ya obtenido con Polars, con la finalidad de comprobar la reproducibilidad del resultado entre dos motores distintos. Una consulta de negocio, sin embargo, no se limita a ordenar: delimita un subconjunto de la población mediante criterios definidos previamente por la gerencia.

En este ejercicio deberá construir una consulta que identifique aquellas empresas que satisfacen **simultáneamente** dos condiciones financieras —capacidad de atender obligaciones de corto plazo y generación de rentabilidad positiva para el accionista— y que presente el resultado ordenado según su exposición al endeudamiento. Este es el procedimiento habitual mediante el cual un área de riesgos delimita una cartera de empresas elegibles.

**Se solicita:**

1. Construir sobre `perfil_reto` una consulta SQL que incluya `SELECT`, `WHERE`, condiciones múltiples enlazadas con `AND` y `ORDER BY`.
2. Establecer como criterios una razón corriente mayor o igual a 1 y un ROE positivo.
3. Ordenar el resultado por endeudamiento, de menor a mayor.
4. Determinar cuántas empresas satisfacen ambos criterios.

**Tiempo estimado:** 10 a 12 minutos.


In [16]:
# Consulta tipo área de riesgos: dos condiciones a la vez y orden por endeudamiento.
consulta_reto4 = duckdb.sql("""
    SELECT empresa,
           ROUND(razon_corriente, 2)      AS razon_corriente,
           ROUND(roe * 100, 1)            AS roe_pct,
           ROUND(endeudamiento * 100, 1)  AS endeudamiento_pct
    FROM perfil_reto
    WHERE razon_corriente >= 1
      AND roe > 0
    ORDER BY endeudamiento_pct ASC
""").df()

display(consulta_reto4)
print("Número de empresas que cumplen ambos criterios:", len(consulta_reto4))


,empresa,razon_corriente,roe_pct,endeudamiento_pct
0,AGROINDUSTRIAL LAREDO S.A.A.,1.10,9.0,34.8
1,CASA GRANDE SOCIEDAD ANONIMA ABIERTA,1.18,7.0,45.4
2,EMPRESA AGROINDUSTRIAL POMALCA S.A.A.,1.55,1.4,50.2


Número de empresas que cumplen ambos criterios: 3


**Pregunta 7.** Según los criterios establecidos, ¿qué empresa presenta la condición financiera más favorable dentro del grupo analizado?

**Respuesta (máximo cuatro líneas):** Agroindustrial Laredo. Cumple los dos criterios y lo hace con el menor endeudamiento del grupo (34.8%) y un ROE competitivo (9.0%). Casa Grande y Pomalca también pasan el filtro, pero con más deuda (45.4% y 50.2%) y, en el caso de Pomalca, con una rentabilidad muy baja (1.4%). Cartavio no aparece pese a liderar el ROE, y lo más probable es que sea por no llegar a una razón corriente de 1.

---



### Ejercicio 5 — Representación conjunta de la rentabilidad y del nivel de endeudamiento

Los gráficos de barras construidos en la Actividad 2 comparan una dimensión por vez. La decisión empresarial, sin embargo, rara vez depende de un solo indicador: una rentabilidad elevada sostenida sobre un endeudamiento igualmente elevado no equivale a una rentabilidad semejante obtenida con una estructura financiera conservadora, aun cuando ambas presenten el mismo ROE.

El gráfico de dispersión permite examinar dos dimensiones de manera simultánea y, mediante el tamaño del marcador, incorporar una tercera. Al situar el endeudamiento en el eje horizontal y el ROE en el eje vertical, cada cuadrante adquiere una lectura gerencial precisa: la combinación de rentabilidad elevada con endeudamiento reducido corresponde al perfil más favorable, mientras que la combinación inversa exige un examen detenido.

**Se solicita:**

1. Construir con Plotly un gráfico de dispersión que sitúe el endeudamiento en el eje horizontal y el ROE en el eje vertical.
2. Identificar cada punto con el nombre de la empresa correspondiente.
3. Representar el activo total mediante el tamaño del marcador.
4. Rotular los ejes y titular el gráfico de modo que resulte interpretable sin recurrir al código.

**Tiempo estimado:** 10 a 12 minutos.


In [17]:
# Dispersión: endeudamiento en X, ROE en Y y el tamaño del punto es el activo total.
fig_disp = go.Figure(go.Scatter(
    x=perfil_reto["endeudamiento"] * 100,
    y=perfil_reto["roe"] * 100,
    mode="markers+text",
    text=perfil_reto["empresa"],
    textposition="top center",
    marker=dict(
        size=perfil_reto["activo_total"],
        sizemode="area",
        sizeref=2. * perfil_reto["activo_total"].max() / (60.**2),
        sizemin=8,
        color="#00A9CE",
        line=dict(width=1, color="white"),
    ),
))
fig_disp.update_layout(
    title="ROE vs. Endeudamiento (tamaño del marcador = activo total) — sector azucarero",
    xaxis_title="Endeudamiento (%)",
    yaxis_title="ROE (%)",
    height=520,
    width=900,
    template="plotly_white",
)
fig_disp.show()

**Pregunta 8.** ¿Qué empresa muestra la combinación más favorable entre rentabilidad y nivel de endeudamiento?

**Respuesta (máximo cuatro líneas):** Agroindustrial Laredo, porque logra un ROE sólido (9%) con la menor deuda del grupo (35%). Cartavio tiene el ROE más alto (24%), pero también el mayor endeudamiento (59%), o sea que esa rentabilidad viene con más riesgo financiero detrás. Casa Grande y Pomalca quedan en el peor lugar: deuda media-alta con ROE bajo. Por eso Laredo ofrece el mejor equilibrio entre rentabilidad y riesgo.

---



### Retroalimentación final — Vinculación con el modelo DELTA Plus

Los cinco ejercicios anteriores se resolvieron mediante Python, DuckDB y Plotly sobre datos financieros oficiales. Corresponde ahora situar ese trabajo dentro del marco conceptual de la sesión. Responda de manera breve.

**Pregunta 9.** A partir del análisis realizado mediante Python, identifique qué dimensión de DELTA Plus se evidencia directamente en esta actividad: *Technology*, *Analytical Techniques* o ambas. Justifique su respuesta en un máximo de tres líneas.

**Respuesta:** Se evidencian las dos. Technology, porque todo el trabajo se apoyó en infraestructura de procesamiento (Python, Polars, DuckDB) y de visualización (Plotly) sobre datos reales. Y Analytical Techniques, porque se aplicaron técnicas descriptivas —rankings e indicadores— y también diagnósticas, como la descomposición DuPont, para explicar el desempeño.

---

**Pregunta 10.** *Technology.* ¿Qué tecnologías resultarían necesarias para que una de las empresas analizadas pudiera ejecutar de manera periódica y reproducible el análisis desarrollado en este laboratorio?

**Respuesta (máximo cuatro líneas):** Tendría que automatizar la descarga desde la API de la SMV en vez de hacerla a mano, contar con un entorno reproducible (notebooks versionados o un pipeline en la nube), un almacenamiento estructurado tipo data warehouse para guardar históricos, y un tablero que actualice los indicadores sin tener que reescribir el código cada vez.

---

**Pregunta 11.** *Analytical Techniques.* ¿En qué nivel de técnicas analíticas se sitúa el análisis realizado en esta sesión y qué técnica adicional podría incorporar la organización para mejorar sus decisiones?

**Respuesta (máximo cuatro líneas):** El análisis se ubica sobre todo en el nivel diagnóstico, porque con DuPont explica por qué se llega a ese ROA, y tiene además una base descriptiva de rankings y comparaciones. Para subir un escalón, la organización podría incorporar técnicas predictivas: por ejemplo proyectar el ROE o el riesgo de liquidez ante distintos escenarios de tipo de cambio o precio del azúcar, usando las series del BCRP.

---



## Conclusiones

Registre tres conclusiones técnicas y de negocio:

1. Los indicadores resistieron la verificación cruzada: Polars y DuckDB devolvieron el mismo ordenamiento y la identidad DuPont se cumplió con una diferencia máxima de ~7×10⁻¹⁸ entre el ROA directo y el ROA reconstruido. Eso confirma que el procedimiento aplicado sobre datos oficiales de la SMV es reproducible y no depende de la herramienta que se use.
2. En el sector aparecen dos perfiles claramente distintos: Cartavio manda en rentabilidad (ROE 24%, ROA 9.9%) apoyada sobre todo en su margen, pero cargando el mayor endeudamiento del grupo (59%); Agroindustrial Laredo, en cambio, sostiene un ROE de alrededor de 9% con la deuda más baja (35%), y termina siendo la de mejor equilibrio entre riesgo y rentabilidad.
3. En términos de DELTA Plus, el laboratorio muestra un uso todavía inicial de Technology y Analytical Techniques, ubicado en el nivel diagnóstico. Para avanzar hacia un estadio superior, las empresas del sector tendrían que automatizar la actualización de estos indicadores e incorporar técnicas predictivas que integren variables sectoriales del BCRP, como el tipo de cambio y las exportaciones.

## Referencias oficiales

- Superintendencia del Mercado de Valores. Portal de Datos Abiertos y servicio de información financiera.
- Banco Central de Reserva del Perú. Series estadísticas mensuales.

